### PR TITLE
Allowing use of magrittr pipe in `survfit2()`

### LINKED ISSUE #93: Allow magrittr data piping into `survfit2()`
Currently, data cannot be piped into `survfit2()` using the magrittr pipe. We'd like to improve upon that!

### PR DESCRIPTION
**Pull request by ddsjoberg**
**What changes are proposed in this pull request?**
* Updating the saved environment in `survfit2()` to include the magrittr `.` when it is present.

**If there is an GitHub issue associated with this pull request, please provide link.**
closes #93 

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark is as complete)

- [x] Ensure all package dependencies are installed by running `renv::install()`
- [x] PR branch has pulled the most recent updates from master branch. Ensure the pull request branch and your local version match and both have the latest updates from the master branch.
- [x] If a new function was added, function included in `_pkgdown.yml`
- [x] If a bug was fixed, a unit test was added for the bug check
- [x] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [x] Overall code coverage remains >99.5%. Review coverage with `withr::with_envvar(new = c("NOT_CRAN" = "true"), covr::report())`. Begin in a fresh R session without any packages loaded. 
- [x] R CMD Check runs without errors, warnings, and notes
- [x] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# ggsurvfit (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")` 
- [ ] Run `codemetar::write_codemeta()`
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

**Comment by shannonpileggi** [comment]
the only thing i could find is that

``` r
library(ggsurvfit)
#> Loading required package: ggplot2

df_colon %>% 
  survfit2(Surv(.$time, .$status) ~ surg, data = .) %>% 
  ggsurvfit(size = 1) 
```



<sup>Created on 2022-10-03 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>


results in the `time` variable label not being passed through

**Comment by ddsjoberg** [comment]
Thanks for taking a look @shannonpileggi !!

File: R/ggsurvfit.R [hunk 1]
```diff
@@ -211,16 +211,15 @@ ggsurvfit <- function(x, type = "survival",
 
   # return time label ----------------------------------------------------------
   switch( # using the CDISC variable as default label, if present
-    isTRUE(any(c("PARAM", "PARAMCD") %in% names(data))),
+    isTRUE(!is.null(data) && any(c("PARAM", "PARAMCD") %in% names(data))),
```
Comment (SHAESEN2) [inline R/ggsurvfit.R:214]: You would need to check two things:
1. is the param/paramcd unique in the data used for estimation. If not, then the programmer forgot to make a subset
2. if param or paramcd was used as strata then label should default to Time

**Review by bailliem — APPROVED** [review]
Is it worth adding an example in survfit2 to explain the current behaviour with pipes i.e. on line 64 add something like 

```
df_lung %>% survfit2(Surv(time, status) ~ sex, data  = .) %>%
  summary(times = c(10, 20))
```

**Comment by ddsjoberg** [comment]
> I feel it would be worth adding an example and text in the survfit2 docstring to explain the current behaviour with pipes i.e. on line 64 add something like
> 
> ```
> df_lung %>% survfit2(Surv(time, status) ~ sex, data  = .) %>%
>   summary(times = c(10, 20))
> ```

Thanks @bailliem ! I think because we're accounting for a non-standard use of the magrittr pipe (typically it's passed to the first argument), we don't need an example highlighting how to use it this way. Also, Hadley is prepping the purrr v1.0 release, and he's removed all `%>%` and replaced them with `|>`.

**Comment by ddsjoberg** [comment]
@SHAESEN2 let's address and PARAM changes in a separate PR

**Pull request merged**